### PR TITLE
Add Spring Starter

### PR DESCRIPTION
**Pull request by SentryMan**
Reflection heavy though it is, you cannot deny that spring is one of the most popular frameworks of all time. This PR adds a module that autoconfigures spring to serialize/deserialize JSON HTTP messages with avaje jsonb. 

tested with https://github.com/spring-guides/gs-spring-boot/tree/main/complete

Didn't do Webflux because I expect reactive APIs to eventually become even more uncommon with the advent of Loom

### CODE REVIEW

File: jsonb-spring-adapter/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports [hunk 1]
```diff
@@ -0,0 +1,2 @@
+io.avaje.jsonb.spring.JsonBAutoConfiguration
```
Comment (rbygrave) [inline jsonb-spring-adapter/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:1]: Is it more consistent to be named JsonbAutoConfiguration ?  
(more consistent with JsonbHttpConverterAutoConfiguration that is)

Comment (SentryMan) [reply jsonb-spring-adapter/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:1]: I'm bad at naming things as previously stated

Comment (SentryMan) [reply jsonb-spring-adapter/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:1]: oh wait yeah the b should be lowercase

Comment (rbygrave) [reply jsonb-spring-adapter/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:1]: yeah sorry, I should have been explicit, I was only referring to the b being lowercase

**Pull request merged**